app/controllers/public_health_controller.rb

```diff
@@ -493,161 +442,62 @@ def advanced_filter(patients, filters)
     end
     patients
   end
-  # rubocop:enable Metrics/MethodLength
-
-  def advanced_filter_enrolled_relative(patients, filter)
-    case filter[:relativeOption]
-    when 'today'
-      patients = patients.where('Date(patients.created_at) = ?', Date.today)
-    when 'tomorrow'
-      patients = patients.where('Date(patients.created_at) = ?', Date.tomorrow)
-    when 'yesterday'
-      patients = patients.where('Date(patients.created_at) = ?', Date.yesterday)
-    when 'custom'
-      case filter[:value][:when]
-      when 'past'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) >= ?', number.days.ago).where('Date(patients.created_at) <= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) >= ?', number.weeks.ago).where('Date(patients.created_at) <= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) >= ?', number.months.ago).where('Date(patients.created_at) <= ?', Date.today)
-        end
-      when 'next'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) <= ?', number.days.from_now).where('Date(patients.created_at) >= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) <= ?', number.weeks.from_now).where('Date(patients.created_at) >= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(patients.created_at) <= ?', number.months.from_now).where('Date(patients.created_at) >= ?', Date.today)
-        end
-      end
+
+  def advanced_filter_date(patients, field, filter, tz_offset)
+    # adjust for difference between client and server timezone (+ instead of - because js and ruby offsets are flipped)
+    tz_diff = tz_offset.to_i.minutes + DateTime.now.utc_offset
```

Comment:
this is for users who aren't in the same timezone as the server, these timestamps/filters are supposed to work/be displayed based on users' timezones

Reply:
Right. I'm wondering why adding DateTime.now.utc_offset is necessary because it will always be 0 since DateTime.now will always be in UTC already since this is on the backend system which is in UTC. The tz_offset already reflects the user timezone offset from UTC.

Reply:
After some offline discussion, we decided to keep this but add a comment. It doesn't actually do anything on prod but if the system was ever in any other timezone than UTC (such as when working locally), it is needed. Resolving.